daemon/api_sideload_n_try.go

```diff
@@ -123,97 +153,156 @@ func sideloadOrTrySnap(c *Command, body io.ReadCloser, boundary string) Response
 		}
 		return trySnap(c.d.overlord.State(), form.Values["snap-path"][0], flags)
 	}
-	flags.RemoveSnapPath = true
 
+	flags.RemoveSnapPath = true
 	flags.Unaliased = isTrue(form, "unaliased")
 	flags.IgnoreRunning = isTrue(form, "ignore-running")
-	systemRestartImmediate := isTrue(form, "system-restart-immediate")
 
-	origPath, tempPath, errRsp := form.SnapFileNameAndTempPath()
+	sideloadFlags := sideloadFlags{
+		Flags:       flags,
+		dangerousOK: isTrue(form, "dangerous"),
+	}
+
+	snapFiles, errRsp := form.GetSnapFiles()
+	if errRsp != nil {
+		return errRsp
+	}
+
+	st := c.d.overlord.State()
+	st.Lock()
+	defer st.Unlock()
+
+	var chg *state.Change
+	if len(snapFiles) > 1 {
+		chg, errRsp = sideloadManySnaps(st, snapFiles, sideloadFlags, user)
+	} else {
+		chg, errRsp = sideloadSnap(st, snapFiles[0], sideloadFlags)
+	}
 	if errRsp != nil {
 		return errRsp
 	}
 
+	systemRestartImmediate := isTrue(form, "system-restart-immediate")
+	if systemRestartImmediate {
+		chg.Set("system-restart-immediate", true)
+	}
```

Comment:
Unless `"system-restart-immediate" == false` has some special meaning, I'd just write:
```suggestion
	chg.Set("system-restart-immediate", isTrue(form, "system-restart-immediate"))
```

Reply:
Changed. Thanks!